### PR TITLE
ARROW-13283: [Archery][Dev] Support passing CPU/memory limits to Docker

### PR DESCRIPTION
**Pull request by lidavidm**
This adds a fairly simplistic way to pass CPU/memory limits to Docker, with default values emulating GitHub actions. Note you'll likely need `ARCHERY_DOCKER_BIN="sudo docker"` unless you have rootless Docker configured with cgroups v2 (else Docker will silently ignore the limits).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -62,6 +62,14 @@ x-ccache: &ccache
   CCACHE_MAXSIZE: 500M
   CCACHE_DIR: /ccache
 
+# CPU/memory limits to pass to Docker. These values emulate GitHub Actions:
+# https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+x-limits:
+  # Note we use cpuset and not cpus since Ninja only detects and limits parallelism
```
Comment (cyb70289) [inline docker-compose.yml:68]: Tested `--cpus` and `--cpuset-cpus`.
With `--cpuset-cpus=0,1`, ninja spawns 2 jobs, as expected.
With `--cpus=2`, ninja spawns 32 jobs on my 32 core machine, though only 2 host cores can be used.

**Review by cyb70289 — APPROVED** [review]
LGTM, +1

File: docker-compose.yml (hunk 1 above)
Comment (lidavidm) [reply docker-compose.yml:68]: For reference, ninja explicitly checks for cpuset: https://github.com/ninja-build/ninja/commit/1bcc689324bdee090eed035353724abc3fa7c909

**Comment by lidavidm** [comment]
@kszucs any thoughts on the CLI flags/YAML added here?

File: docker-compose.yml (hunk 1 above)
Comment (kszucs) [reply docker-compose.yml:68]: If I understand correctly we can't set the limits manually, just use preset(s) defined in the compose file?
If so we may want to define multiple presets in the future, e.g. big-memory, small-memory, travis, etc. 

How about a `--resource-limit=github` or `--limit-preset=github` or something similar on the CLI and a mapping under `x-limit-presets` in the compose file?

Comment (lidavidm) [reply docker-compose.yml]: Thanks - updated it to allow defining multiple presets, though there's still only the GitHub one defined by default.

**Review by kszucs — APPROVED** [review]
Thanks David!

**Pull request closed without merging**